RNTester/src/RNTesterApp.re

```diff
@@ -31,41 +38,52 @@ let header = (~onBack=?, ~title, ()) =>
       switch onBack {
       | None => ReasonReact.nullElement
       | Some(onBack) =>
-        <View style=styles##headerLeft> <Button title="Back" onPress=onBack /> </View>
+        <View style=styles##headerLeft>
+          <Button title="Back" onPress=onBack />
+        </View>
       }
     )
   </View>;
 
+type action =
+  | ChangeCurrentExample(option(ExampleList.item));
+
 type state = {currentExample: option(ExampleList.item)};
 
-let component = ReasonReact.statefulComponent("RNTesterApp");
+let component = ReasonReact.reducerComponent("RNTesterApp");
 
-let make = (_children) => {
-  let onPress = (item, {ReasonReact.state}) =>
+let make = _children => {
+  let onPress = (item, {ReasonReact.state, ReasonReact.send}) =>
     switch state.currentExample {
-    | None => ReasonReact.Update({currentExample: Some(item)})
-    | Some(_) => ReasonReact.Update({currentExample: None})
+    | None => send(ChangeCurrentExample(Some(item)))
+    | Some(_) => send(ChangeCurrentExample(None))
     };
-  let onBack = ((), _self) => ReasonReact.Update({currentExample: None});
+  let onBack = (_event, {ReasonReact.send}) =>
+    send(ChangeCurrentExample(None));
   {
     ...component,
     initialState: () => {currentExample: None},
-    render: ({state, update}) => {
+    reducer: (action, _state) =>
+      switch action {
+      | ChangeCurrentExample(example) => Update({currentExample: example})
+      },
+    render: ({state, handle}) => {
       let components = ExampleList.components;
       switch state.currentExample {
       | None =>
         <View style=styles##exampleContainer>
           (header(~title="ReasonRNTester", ()))
-          <RNTesterExampleList components onPress=(update(onPress)) />
+          <RNTesterExampleList components onPress=(handle(onPress)) />
```

Comment:
~~We should use `reduce` here.~~ I forgot it's deprecated. How about an inline function calling send?

Reply:
✔️ done

however, is there any reason why we should use inline functions?

Reply:
The result JS is better because there's no currying. Also, I find the indirection with handle a little bit weird. For me handle says, this is a function which performs side effects. Although with `reduce` changed to `send` the difference is a bit blurry.